src/protocolsupport/protocol/typeremapper/basic/TileNBTRemapper.java

```diff
@@ -90,28 +136,142 @@ protected static void register(TileEntityType type, BiFunction<ProtocolVersion,
 			},
 			ProtocolVersionsHelper.BEFORE_1_9
 		);
-		register(
-			TileEntityType.SKULL,
-			(version, input) -> {
-				//TODO
+		List<Material> skulls = Arrays.asList(
+				Material.SKELETON_SKULL, 
+				Material.WITHER_SKELETON_SKULL, 
+				Material.CREEPER_HEAD, 
+				Material.DRAGON_HEAD, 
+				Material.PLAYER_HEAD, 
+				Material.ZOMBIE_HEAD,
+				Material.SKELETON_WALL_SKULL, 
+				Material.WITHER_SKELETON_WALL_SKULL, 
+				Material.CREEPER_WALL_HEAD, 
+				Material.DRAGON_WALL_HEAD, 
+				Material.PLAYER_WALL_HEAD, 
+				Material.ZOMBIE_WALL_HEAD
+		);
+		registerLegacy(
+			skulls, (blockdata, input) -> {
+				byte skulltype = 0;
+				switch(blockdata.getMaterial()) {
+					case SKELETON_SKULL: case SKELETON_WALL_SKULL: skulltype = 0; break;
+					case WITHER_SKELETON_SKULL: case WITHER_SKELETON_WALL_SKULL: skulltype = 1; break;
+					case ZOMBIE_HEAD: case ZOMBIE_WALL_HEAD: skulltype = 2; break;
+					case PLAYER_HEAD: case PLAYER_WALL_HEAD: skulltype = 3; break;
+					case CREEPER_HEAD: case CREEPER_WALL_HEAD: skulltype = 4; break;
+					case DRAGON_HEAD: case DRAGON_WALL_HEAD: skulltype = 5; break;
+					default: break;
+				}
+				input.setTag("SkullType", new NBTByte(skulltype));
+				if (blockdata instanceof Rotatable) {
+					System.out.println("GOIND FOR ORIENTATION REMAP!");
```

Comment:
Remove debug print.

IDEs allow to search for text in files. Do a search for debug prints before PRing anything.